File: SearchA2DMatrix/SearchA2DMatrix.java
import java.io.*;
import java.util.*;

class SearchA2DMatrix {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int rows = scanner.nextInt();
        int columns = scanner.nextInt();

        int[][] matrix = new int[rows][columns];
        for(int i = 0 ; i < rows ; ++i) {
            for(int j = 0 ; j < columns ; ++j) {
                matrix[i][j] = scanner.nextInt();
            }
        }
        int queries = scanner.nextInt();

        for(int i = 0 ; i < queries ; ++i) {
            int target = scanner.nextInt();
            boolean result = new Solution().searchMatrix(matrix , target);
            System.out.println(result);
        }
        scanner.close();
    }
}

File: SearchA2DMatrix/SearchA2DMatrix.py
import sys
from Solution import solve


def main():
    rows, columns = map(int, sys.stdin.readline().split())
    matrix = list()
    for _ in range(rows):
        row = list(map(int, sys.stdin.readline().split()))
        matrix.append(row)
    queries = int(sys.stdin.readline())
    for _ in range(queries):
        target = int(sys.stdin.readline())
        result = solve(matrix, target)
        if result:
            sys.stdout.write("true\n")
        else:
            sys.stdout.write("false\n")


if __name__ == "__main__":
    main()
